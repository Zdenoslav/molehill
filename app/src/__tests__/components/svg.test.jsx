import React from 'react';
import renderer from 'react-test-renderer'
import 'jest-styled-components';

import Svg from 'app/components/svg';

describe('Svg', () => {

  it('should render Svg with size', () => {
    const wrapper = renderer.create(<Svg size={10} />).toJSON();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toHaveStyleRule('width', '10px');
    expect(wrapper).toHaveStyleRule('min-width', '10px');
    expect(wrapper).toHaveStyleRule('height', '10px');
    expect(wrapper).toHaveStyleRule('min-height', '10px');
  });

  it('should render Svg with width and height', () => {
    const wrapper = renderer.create(<Svg width={5} height={2} />).toJSON();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toHaveStyleRule('width', '5px');
    expect(wrapper).toHaveStyleRule('min-width', '5px');
    expect(wrapper).toHaveStyleRule('height', '2px');
    expect(wrapper).toHaveStyleRule('min-height', '2px');
  });
});
