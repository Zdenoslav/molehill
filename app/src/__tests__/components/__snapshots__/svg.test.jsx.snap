// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Svg should render Svg 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
      width: ",
          [Function],
          ";
      min-width: ",
          [Function],
          ";
      height: ",
          [Function],
          ";
      min-height: ",
          [Function],
          ";
    ",
        ],
      },
      "displayName": "styled.svg",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "svg",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Svg should render Svg with size 1`] = `
.c0 {
  width: 10px;
  min-width: 10px;
  height: 10px;
  min-height: 10px;
}

<svg
  className="c0"
  size={10}
/>
`;

exports[`Svg should render Svg with width and height 1`] = `
.c0 {
  width: 5px;
  min-width: 5px;
  height: 2px;
  min-height: 2px;
}

<svg
  className="c0"
  height={2}
  width={5}
/>
`;
